tests/contracts/equal.clar

```diff
@@ -53,3 +53,147 @@
 (define-public (uint-unequal-2)
     (ok (is-eq u43 u43 u54 u56 u43 u43))
 )
+
+(define-public (buf-equal)
+    (ok (is-eq 0x0102 0x0102))
+)
+
+(define-public (buf-equal-2)
+    (ok (is-eq 0x0102 0x0102 0x0102))
+)
+
+(define-public (buf-unequal)
+    (ok (is-eq 0x0102 0x0103))
+)
+
+(define-public (buf-unequal-2)
+    (ok (is-eq 0x0102 0x010203))
+)
+
+(define-public (buf-unequal-3)
+    (ok (is-eq 0x01 0x01 0x02))
+)
+
+(define-public (str-ascii-equal)
+    (ok (is-eq "hello" "hello"))
+)
+
+(define-public (str-ascii-unequal)
+    (ok (is-eq "hello" "world"))
+)
+
+(define-public (principal-equal)
+    (ok (is-eq 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM))
+)
+
+(define-public (principal-unequal)
+    (ok (is-eq 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR))
+)
+
+(define-public (call-principal-equal)
+    (ok (is-eq 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.foo 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.foo))
+)
+
+(define-public (call-principal-unequal)
+    (ok (is-eq 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.foo 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.bar))
+)
+
+(define-public (call-principal-unequal-2)
+    (ok (is-eq 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.foo 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR.foo))
+)
+
+(define-public (call-optional-equal)
+    (ok (is-eq (some 1) (some 1)))
+)
+
+(define-public (call-optional-unequal)
+    (ok (is-eq (some 0x01) (some 0x02)))
+)
+
+(define-public (call-response-ok-equal)
+    (ok (is-eq (ok 0) (ok 0)))
+)
+
+(define-public (call-response-err-equal)
+    (ok (is-eq (err 0x010203) (err 0x010203)))
+)
+
+(define-public (call-response-ok-err-unequal)
+    (ok (is-eq (ok 42) (err "forty-two")))
+)
+
```

Comment:
How about adding a test comparing an ok and an err with the same value?

```clarity
(is-eq (ok 42) (err 42))
```

Reply:
added in 93aa27cdcbd93d5a5b3ccbbd7c26aacc5f47f195